_posts/2018-01-07-building-serverless-radiator.md

```diff
@@ -0,0 +1,225 @@
+
+AWS Lambda environment currently supports Node.js, Java, C# and Python languages. For an endpoint exposer I chose Python which have excellent [Boto 3](https://boto3.readthedocs.io/en/latest/) SDK for accessing AWS.
+
+First I started from what I want as a result. The result should be JSON response with different keys for alarms, pipelines and metrics. 
+
+```json
+{
+    "alarms": [
+        {
+            "AlarmName": "product-name-service-500",
+            "StateValue": "OK"
+        }
+    ],
+    "pipelines": [
+        {
+            "name": "ProjectInfraBuildPipeline",
+            "currentStatus": "Succeeded"
+        }
+    ],
+    "metrics": [
+        {
+            "name": "RDS CPU Utilization",
+            "statistics": "Average",
+            "unit": "Percent",
+            "result": 7.004
+        }
+    ]
+}
+```
+
+The main handler is therefore quite simple. There is no need to use any parameters because we return always all the pipelines and alarms.   
+
+```python
+def status(event, context):
+    result = {
+        "alarms":get_alarms(),
+        "pipelines" : get_pipelines(),
+        "metrics" : get_metrics()}
+    return {"body": json.dumps(result)}
+```
+
+Getting pipeline statuses and alarms is quite straightforward. The alarms can be fetched with one API call. Pipelines need to be first fetched and then current statuses can be checked one by one. For the radiator I also formatted the result a little bit.
+
+Metrics cannot be fetched with one call because AWS has tons of different kind metrics available. Currently I make a Python data structure which describes what kind of metric I want to expose. The next code example gets average RDS CPU utilization of last 10 minutes.
+
+```python
+metrics= [{'name': 'RDS CPU Utilization',
+            'request': {'Namespace':'AWS/RDS',
+                         'MetricName':'CPUUtilization',
+                         'StartTime': datetime.utcnow() - timedelta(minutes=10),
+                         'EndTime': datetime.now() ,
+                         'Period': 600,
+                         'Statistics':['Average'],
+                         'Unit':'Percent'},
+            'statistics': 'Average',
+            'unit': 'Percent'}]
+```            
+
+Lambda can be installed by hand via AWS Console but automatized installing is much more preferable. Fortunately [Serverless Framework](https://serverless.com/) supports also deploying Python code. 
+
+With Serverless installing the Radiator Exposer is easy. Just run ``serverless deploy``. Serverless  runs for a while and returns a new endpoint and an api key necessary for authentication. 
```

Comment:
"Installing the Radiator Exposer is easy with Serverless" kuulostaa korvaani paremmalta sanajärjestykseltä.

"Serverless  runs" <- kaksi välilyöntiä
  
edit: Tän pätkän sanoman mielestäni voisi siirtää "Building a radiator" otsikon alle.